roombasim/pittras/task/hold_position_task.py

```diff
@@ -0,0 +1,92 @@
+'''
+hold_position_task.py
+'''
+
+import numpy as np
+
+import roombasim.config as cfg
+
+from roombasim.ai import Task, TaskState
+
+class HoldPositionTaskStates:
+    '''
+    States for the Hold Position task.
+    '''
+    init = 0
+    holding = 1
+    done = 2
+    failed = 3
+
+class HoldPositionTask(Task):
+    '''
+    A task to hold the drone at its current absolute position for some duration
+    of time. The task accepts a float64 that specifies such duration in seconds,
+    which means holding the current position indefinitely if less than or equal
+    to zero.
+    '''
+
+    def __init__(self, hold_duration):
+        self.hold_duration = hold_duration
+        self.state = HoldPositionTaskStates.init
+
+        # Hold position properties
+        self.start_time = 0
+        self.hold_xy = np.array([0,0])
+        self.hold_z = 0
+
+        # PID controller contstants
+        self.k_xy = cfg.PITTRAS_PID_XY
+        self.k_z = cfg.PITTRAS_PID_Z
+
+        self.i_xy = np.array([0,0], dtype=np.float64)
+        self.i_z = 0
+
+    def update(self, delta, elapsed, state_controller, environment):
+        if (self.state == HoldPositionTaskStates.done or
+                self.state == HoldPositionTaskStates.failed):
+            return
+
+        # Fetch current odometry
+        drone_state = state_controller.query('DroneState', environment)
+
+        if self.state == HoldPositionTaskStates.init:
+            # TODO(zacyu): Remove this when `elapsed` is changed to represent
+            #              the elipsed time since the start of the task.
+            self.start_time = elapsed
+            self.state = HoldPositionTaskStates.holding
+            self.hold_xy = drone_state['xy_pos']
+            self.hold_z = drone_state['z_pos']
+            return
+
+        if (self.hold_duration > 0) and (elapsed - self.start_time >
+                                         self.hold_duration * 1000):
+            if (drone_state['xy_pos'] == self.hold_xy and
```

Comment:
This comparison does not work, you probably want to check if the distance between the vectors is less than some epsilon. E.g: `np.linalg.norm(np.subtract(drone_state['xy_pos'], self.hold_xy)) < epsilon`.

In the future, I would like to add the ability to add noise to sensor data to more accurately model real-world uncertainty. In that case, it is very unlikely that we would get exact floating point accuracy.

Reply:
That's right. It was also mentioned above and I forgot to change it.

Reply:
Shall I just use your `PITTRAS_XYZ_TRANSLATION_ACCURACY` for epsilon? Or perhaps it would be better to rename it to something more generic?

Reply:
@hgarrereyn ping